update/process/baseProcess.go

```diff
@@ -0,0 +1,257 @@
+			Type:            body.MiniBlocks[i].Type,
+		}
+	}
+
+	return totalTxCount, miniBlockHeaders, nil
+}
+
+func (b *baseProcessor) saveAllBlockDataToStorageForSelfShard(
+	headerHandler data.HeaderHandler,
+	body *block.Body,
+) {
+	if check.IfNil(headerHandler) {
+		log.Warn("SaveAllBlockDataToStorageForSelfShard", "error", update.ErrNilHeaderHandler)
+		return
+	}
+	if body == nil {
+		log.Warn("SaveAllBlockDataToStorageForSelfShard", "error", update.ErrNilBlockBody)
+		return
+	}
+	if headerHandler.GetShardID() != b.selfShardID {
+		return
+	}
+
+	miniBlockHeadersHashes := headerHandler.GetMiniBlockHeadersHashes()
+	mapBlockTypesTxs := make(map[block.Type]map[string]data.TransactionHandler)
+	for i := 0; i < len(body.MiniBlocks); i++ {
+		miniBlock := body.MiniBlocks[i]
+		if _, ok := mapBlockTypesTxs[miniBlock.Type]; !ok {
+			mapBlockTypesTxs[miniBlock.Type] = b.txCoordinator.GetAllCurrentUsedTxs(miniBlock.Type)
+		}
+
+		marshalizedMiniBlock, errNotCritical := b.marshalizer.Marshal(miniBlock)
+		if errNotCritical != nil {
+			log.Warn("SaveAllBlockDataToStorageForSelfShard.Marshal -> MiniBlock",
+				"error", errNotCritical.Error())
+			continue
+		}
+
+		errNotCritical = b.storage.Put(dataRetriever.MiniBlockUnit, miniBlockHeadersHashes[i], marshalizedMiniBlock)
+		if errNotCritical != nil {
+			log.Warn("SaveAllBlockDataToStorageForSelfShard.Put -> MiniBlockUnit",
+				"error", errNotCritical.Error())
+		}
+	}
+
+	marshalizedReceipts, errNotCritical := b.txCoordinator.CreateMarshalizedReceipts()
+	if errNotCritical != nil {
+		log.Warn("SaveAllBlockDataToStorageForSelfShard.CreateMarshalizedReceipts",
+			"error", errNotCritical.Error())
+	} else {
+		if len(marshalizedReceipts) > 0 {
+			errNotCritical = b.storage.Put(dataRetriever.ReceiptsUnit, headerHandler.GetReceiptsHash(), marshalizedReceipts)
+			if errNotCritical != nil {
+				log.Warn("SaveAllBlockDataToStorageForSelfShard.Put -> ReceiptsUnit",
+					"error", errNotCritical.Error())
+			}
+		}
+	}
+
+	mapTxs := b.importHandler.GetTransactions()
+	for _, miniBlock := range body.MiniBlocks {
+		for _, txHash := range miniBlock.TxHashes {
+			tx, ok := mapTxs[string(txHash)]
+			if !ok {
+				log.Warn("SaveAllBlockDataToStorageForSelfShard",
+					"error", update.ErrTransactionNotFoundInImportedMap)
+				continue
+			}
+
+			unitType := getUnitTypeFromMiniBlockType(miniBlock.Type)
+			marshaledData, errNotCritical := b.marshalizer.Marshal(tx)
+			if errNotCritical != nil {
+				log.Warn("SaveAllBlockDataToStorageForSelfShard.Marshal -> Transaction",
+					"error", errNotCritical.Error())
+				continue
+			}
+
+			errNotCritical = b.storage.Put(unitType, txHash, marshaledData)
+			if errNotCritical != nil {
+				log.Warn("SaveAllBlockDataToStorageForSelfShard.Put -> Transaction",
+					"error", errNotCritical.Error())
+			}
+		}
+	}
+}
+
+func getUnitTypeFromMiniBlockType(mbType block.Type) dataRetriever.UnitType {
+	unitType := dataRetriever.TransactionUnit
+	switch mbType {
```

Comment:
I think we should have a stricter switch: please add a default that will output an error, otherwise the function will return a wrong storage unit if the type is not defined here (example, by mistake we can provide a peer change miniblock here)

Reply:
done